streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -59,7 +58,7 @@
         "No more records will be sent and no more offsets will be recorded for this task. " +
         "Enable TRACE logging to view failed record key and value.";
     private final static String EXCEPTION_MESSAGE = "%sAbort sending since %s with a previous record (key %s value %s timestamp %d) to topic %s due to %s";
-    private final static String PARAMETER_HINT = "\nYou can increase producer parameter `retries` and `retry.backoff.ms` to avoid this error.";
+    private final static String PARAMETER_HINT = "\nYou can increase the producer config `delivery.timeout.ms` to avoid this error.";
```

Comment:
I think we should change this and mention `TimeoutException` and `deliver.timeout` as a special case. Or maybe better, have two different HINT, one for `TimeoutException` and one for all other `RetriableException`s?

Reply:
I'd +1 for hinting differently for TimeoutException and other retriables.

Reply:
Not sure if you saw the rest of the discussion. Please be specific about what message you would want for other retriables and why.

Reply:
Sure. I'd suggest that around line 220 in RecordCollectorImpl, we catch 1) TimeoutException, and 2) RetriableException, and 3) ProducerFencedException, and 4) all other exceptions separately and log a different error message on those four cases.

For 1) usually users did not override `retries`, so we'd just suggest users to tune `deliver.timeout` while keeping obsoleted `retries` with their default value, and for 2) we'd suggest users to check if they've overridden `retries` and if yes do not do that and tune `deliver.timeout` instead.